docs/uds-package-practices.md

```diff
@@ -0,0 +1,60 @@
+# UDS Package Practices
+
+This document describes the practices that a UDS package **must**, **should**, and **may** follow to be considered officially supported as a UDS package and to be able to display the "Made for UDS" badge on its GitHub repository README.md file:
+
+[<img alt="Made for UDS" src="./made-for-uds.svg" height="20px"/>](https://github.com/defenseunicorns/uds-core)
+
+> [!NOTE]
+> This badge should link to the `uds-core` repo and should match the height of any other badges (i.e. `20px`).
+
+## Integrations
+
+Below are the minimal services that a UDS package **must** integrate with:
+
+### Istio
+
+- **Must** define any external interfaces under the `expose` key.
+- **Must** deploy and operate successfully with Istio injection enabled in the namespace.
+- **Should** avoid workarounds such as disabling strict mTLS peer authentication.
+
+### Network Policies
+
+- **Must** define network policies under the `allow` key as required.
+- **Should** minimize network policies to specific selectors needed for Ingress/Egress traffic.
+- **May** template network policy keys to provide flexibility for delivery customers to configure.
+
+### Keycloak
+
+- **Must** create a client through the `sso` key and use that client for any provided user login.
```

Comment:
Tried covering that with `any provided user login` - i.e. if the tool doesn't provide a way for a user to login (like GLR) than there would be no keycloak integration.  Will think through ways to reword though...

Reply:
```suggestion
- **Must** protect any external endpoints with a client through the `sso` key and use that client for any provided user login.
```

Is this too aggressive? I think in general you'd always want keycloak auth, even if it's with Authservice (want to be careful not to say "if app implements SSO"). There may still be some public external endpoints though...static docs stuff?

Reply:
refined the verbiage a bit

Reply:
I think this is good from my perspective - once Authservice rolls out it might make sense to clarify/restrict this even more to expect authservice integration in some cases.